packages/client-core/src/components/Settings/CheckboxItem.stories.tsx

```diff
@@ -0,0 +1,179 @@
+      }
+    }
+  },
+  tags: ['autodocs'],
+  argTypes: {
+    label: {
+      control: 'text',
+      description: 'The label text for the checkbox'
+    },
+    checked: {
+      control: 'boolean',
+      description: 'Whether the checkbox is checked'
+    },
+    disabled: {
+      control: 'boolean',
+      description: 'Whether the checkbox is disabled'
+    },
+    onClick: {
+      action: 'clicked',
+      description: 'Callback function when checkbox is clicked'
+    }
+  },
+  args: {
+    label: 'I agree to the Infinite Reality Terms of Service',
+    checked: false,
+    disabled: false
+  }
+} satisfies Meta<typeof CheckboxItem>
+
+export default meta
+type Story = StoryObj<typeof meta>
+
+const CheckboxItemRenderer = (args: CheckboxItemProps) => {
+  const [currentArgs, updateArgs] = useArgs<{ checked: boolean }>()
+
+  return (
+    <div className="w-96 rounded-lg bg-gradient-to-br from-pink-500 via-purple-500 to-orange-400 p-4">
+      <CheckboxItem
+        {...args}
+        checked={currentArgs.checked}
+        onClick={() => updateArgs({ checked: !currentArgs.checked })}
+      />
+    </div>
+  )
+}
+
+export const Default: Story = {
+  name: 'Default',
+  render: CheckboxItemRenderer
+}
+
+export const Checked: Story = {
+  name: 'Checked',
+  args: {
+    checked: true
+  },
+  render: CheckboxItemRenderer
+}
+
+export const Disabled: Story = {
+  name: 'Disabled',
+  args: {
+    disabled: true
+  },
+  render: CheckboxItemRenderer
+}
+
+export const DisabledChecked: Story = {
+  name: 'Disabled Checked',
+  args: {
+    checked: true,
+    disabled: true
+  },
+  render: CheckboxItemRenderer
+}
+
+export const AgeVerification: Story = {
+  name: 'Age Verification',
+  args: {
+    label: 'I am 18 years of age or older'
+  },
+  render: CheckboxItemRenderer
+}
+
+export const WithCustomContent: Story = {
+  name: 'With Custom Content',
+  render: (args) => {
+    const [currentArgs, updateArgs] = useArgs<{ checked: boolean }>()
+
+    return (
+      <div className="w-96 rounded-lg bg-gradient-to-br from-pink-500 via-purple-500 to-orange-400 p-4">
+        <CheckboxItem
+          {...args}
+          checked={currentArgs.checked}
+          onClick={() => updateArgs({ checked: !currentArgs.checked })}
+        >
+          <div className="flex flex-col">
+            <span className="font-medium">Custom Content</span>
+            <span className="text-sm text-white/70">This checkbox has custom children content</span>
+          </div>
+        </CheckboxItem>
+      </div>
+    )
+  }
+}
+
+export const MultipleItems: Story = {
+  name: 'Multiple Items',
+  render: () => {
+    const [termsChecked, setTermsChecked] = React.useState(false)
+    const [ageChecked, setAgeChecked] = React.useState(false)
+    const [newsletterChecked, setNewsletterChecked] = React.useState(true)
+
+    return (
+      <div className="w-96 space-y-2 rounded-lg bg-gradient-to-br from-pink-500 via-purple-500 to-orange-400 p-4">
```

Comment:
Setting an exact width means the design will break down in certain cases. Better to use `w-full` with `min-w-96` or `max-w-96`. But this shouldnt need a forced width if the checkboxes have content anyways